saver/src/bin/saver.rs

```diff
@@ -0,0 +1,203 @@
+        write_column!(row_group_writer.next_column()?, end_times, parquet::column::writer::ColumnWriter::ByteArrayColumnWriter);
+
+        let rest_times: Vec<ByteArray> = messages.iter().map(|m| ByteArray::from(m.rest_time.as_str())).collect();
+        write_column!(row_group_writer.next_column()?, rest_times, parquet::column::writer::ColumnWriter::ByteArrayColumnWriter);
+
+        let latitude_starts: Vec<f64> = messages.iter().map(|m| m.latitude_start).collect();
+        write_column!(row_group_writer.next_column()?, latitude_starts, parquet::column::writer::ColumnWriter::DoubleColumnWriter);
+
+        let longitude_starts: Vec<f64> = messages.iter().map(|m| m.longitude_start).collect();
+        write_column!(row_group_writer.next_column()?, longitude_starts, parquet::column::writer::ColumnWriter::DoubleColumnWriter);
+
+        let timestamp_starts: Vec<ByteArray> = messages.iter().map(|m| ByteArray::from(m.timestamp_start.as_str())).collect();
+        write_column!(row_group_writer.next_column()?, timestamp_starts, parquet::column::writer::ColumnWriter::ByteArrayColumnWriter);
+
+        let latitude_ends: Vec<f64> = messages.iter().map(|m| m.latitude_end).collect();
+        write_column!(row_group_writer.next_column()?, latitude_ends, parquet::column::writer::ColumnWriter::DoubleColumnWriter);
+
+        let longitude_ends: Vec<f64> = messages.iter().map(|m| m.longitude_end).collect();
+        write_column!(row_group_writer.next_column()?, longitude_ends, parquet::column::writer::ColumnWriter::DoubleColumnWriter);
+
+        let timestamp_ends: Vec<ByteArray> = messages.iter().map(|m| ByteArray::from(m.timestamp_end.as_str())).collect();
+        write_column!(row_group_writer.next_column()?, timestamp_ends, parquet::column::writer::ColumnWriter::ByteArrayColumnWriter);
+
+        let latitude_rests: Vec<f64> = messages.iter().map(|m| m.latitude_rest).collect();
+        write_column!(row_group_writer.next_column()?, latitude_rests, parquet::column::writer::ColumnWriter::DoubleColumnWriter);
+
+        let longitude_rests: Vec<f64> = messages.iter().map(|m| m.longitude_rest).collect();
+        write_column!(row_group_writer.next_column()?, longitude_rests, parquet::column::writer::ColumnWriter::DoubleColumnWriter);
+
+        let timestamp_rests: Vec<ByteArray> = messages.iter().map(|m| ByteArray::from(m.timestamp_rest.as_str())).collect();
+        write_column!(row_group_writer.next_column()?, timestamp_rests, parquet::column::writer::ColumnWriter::ByteArrayColumnWriter);
+
+        writer.close_row_group(row_group_writer)?;
+    }
+
+    writer.close()?;
+    println!("Parquet file saved to {}", parquet_file);
+    Ok(())
+}
+
+/// Upload the Parquet file to MinIO
+async fn upload_to_minio(
+    bucket: &str,
+    file_path: &str,
+    key: &str,
+    endpoint: &str,
+    access_key: &str,
+    secret_key: &str,
+) -> Result<(), Box<dyn std::error::Error>> {
+    let config = aws_sdk_s3::Config::builder()
```

Comment:
in a next PR, you could refactor this by using https://docs.rs/object_store/latest/object_store/aws/struct.AmazonS3Builder.html
this lib will allow you to write directely your dataframe into your minio
with a .write()